core/dbt/parser/schemas.py

```diff
@@ -351,6 +352,23 @@ def create_test_node(
         column_name: Optional[str],
     ) -> ParsedSchemaTestNode:
 
+        HASH_LENGTH = 10
+
+        def get_hashable_md(
+            data: Union[str, int, float, List, Dict]
+        ) -> Union[str, List, Dict]:
+            if type(data) == dict:
+                return {k: get_hashable_md(data[k]) for k in sorted(data.keys())}  # type: ignore
```

Comment:
What's the logic behind sorting here? We're returning a dictionary which will be unordered.

Reply:
Dicts in python are [**insertion ordered** since 3.6](https://docs.python.org/3.6/whatsnew/3.6.html#new-dict-implementation) (that's why we're ok with returning a dict as a final output of the function).  That said, I didn't have much confidence that the order would be consistent given it's generated... somewhere else.  It would also be an easy thing to break because of that so I added an extra sort.

Reply:
This might make it clearer:
```
> /Users/iknox/.pyenv/versions/dbt-dev/lib/python3.8/site-packages/dbt/parser/schemas.py(381)create_test_node()
(Pdb) dct_1 = {'c':'C', 'a':'A', 'b':'B'}
(Pdb) dct_2 = {'b':'B', 'a':'A', 'c':'C'}
(Pdb) (get_hashable_md(dct_1), get_hashable_md(dct_2))
({'a': 'A', 'b': 'B', 'c': 'C'}, {'a': 'A', 'b': 'B', 'c': 'C'})
(Pdb) (get_hashable_md_no_keysort(dct_1), get_hashable_md_no_keysort(dct_2))
({'c': 'C', 'a': 'A', 'b': 'B'}, {'b': 'B', 'a': 'A', 'c': 'C'})
```

Reply:
I didn't know that! Good fact to file away for later. And because we're using it as a string later, the order matters here. It does mean that if they swap dict implementations in a future version of python this might break on a version upgrade. But I'm not actually that worried about this.

Reply:
Yeah, it's been this way since 3.6 and documented as a language feature since 3.7-- I don't see it disappearing any time soon.